tel2puml/otel_to_puml.py

```diff
@@ -0,0 +1,90 @@
+"""Module to handle e2e operations of handling otel data and producing puml
+files."""
+
+import os
+from typing import Optional, Literal
+
+from tel2puml.tel2puml_types import OtelPumlOptions, PVPumlOptions
+from tel2puml.otel_to_pv.sequence_otel_v2 import otel_to_pv
+from tel2puml.pv_to_puml import (
+    pv_streams_to_puml_files,
+    pv_files_to_pv_streams,
+)
+from tel2puml.otel_to_pv.config import load_config_from_yaml
+from tel2puml.otel_to_pv.ingest_otel_data import ingest_data_into_dataholder
+from tel2puml.otel_to_pv.data_holders import (
+    DataHolder,
+)
+
+
+def otel_to_puml(
+    otel_to_puml_options: Optional[OtelPumlOptions] = None,
+    pv_to_puml_options: Optional[PVPumlOptions] = None,
+    output_file_directory: str = "puml_output",
+    components: Literal["all", "otel_to_puml", "pv_to_puml"] = "all",
+) -> DataHolder | None:
+    """Creates puml files from otel data. Takes optional parameters to handle
+    separate parts of the process individually.
+
+    :param otel_to_puml_options: Options for otel to puml. Defaults to `None`
+    :type otel_to_puml: `Optional`[:class:`OtelPumlOptions`]
+    :param pv_to_puml_options: Options for pv to puml. Defaults to `None`
+    :type pv_to_puml_options: `Optional`[:class:`PVPumlOptions`]
+    :param output_file_directory: Output file directory. Defaults to
+    "puml_output".
+    :type output_file_directory: `str`
+    :param components: The parts of the process that are required. Defaults to
+    "all".
+    :type components: `Literal`["all", "otel_to_puml", "pv_to_puml"]
+    :return: DataHolder object or None
+    :rtype: :class:`DataHolder` | `None`
+    """
+    # Create output directory if non-existent
+    if not os.path.isdir(output_file_directory):
+        os.mkdir(output_file_directory)
+
+    # Handle different components
+    if components == "all":
```

Comment:
I think use a match statement and group all and otel_to_pv together. Then just have an if statement for otel_to_pv option to return after calling otel_to_pv as if we just select otel_to_pv we just want to be doing the stuff just for otel to pv